internal/controller/config/config.go

```diff
@@ -92,10 +66,12 @@ func (cm *ConfigManager) LoadConfig(onErrorReloading func(error), configPath, co
 	v.OnConfigChange(func(e fsnotify.Event) {
 		cm.lock.Lock()
 		defer cm.lock.Unlock()
-		err = v.Unmarshal(cm.config)
+		newConfig := GlobalConfig{}
+		err = v.Unmarshal(&newConfig)
 		if err != nil {
 			onErrorReloading(err)
 		}
+		cm.config = &newConfig
```

Comment:
when you call `v.Unmarshal()` it doesn't first clear out the memory of the original object, so if you use the same memory as before, the old fields can still be in there...therefore creating new memory

Reply:
Good catch, do we have a test case that will fail without your change?

Reply:
doing that now

Reply:
just added one, and tried the same test using the "main" branch to verify it would fail

Reply:
darn, need to see why it's failing the race test in the CI...

Reply:
fixed the race condition, which was caused by calling `v.OnConfigChange()` after `v.WatchConfig()`